modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/http/SetStatusCode.java

```diff
@@ -52,8 +54,12 @@
     public BValue[] execute(Context context) {
         try {
             BMessage bMsg = (BMessage) getArgument(context, 0);
-            int statusCode = ((BInteger) getArgument(context, 1)).intValue();
-            bMsg.value().setProperty(Constants.HTTP_STATUS_CODE, statusCode);
+            long statusCode = ((BInteger) getArgument(context, 1)).intValue();
+            if (statusCode != (int) statusCode) {
+                throw BLangExceptionHelper
+                        .getRuntimeException(RuntimeErrors.INDEX_NUMBER_TOO_LARGE, statusCode); //todo, is this right?
```

Comment:
need to discuss whether this is right, hence the todo. anyway is this right? even though practically status codes are integer values, is it correct to limit?